### PR TITLE
Add GitHub release

### LINKED ISSUE #251: Push PS Gallery versions also to GitHub Releases
Would be nice if the releases are sent to GitHub releases as well.
I like "watching" releases, and I think it's also good for the project when you can see regular releases on GH releases

It can be automated in Azure Pipelines

### PR DESCRIPTION
**Pull request by MichelZ**
# PR Summary

Add the release to GitHub as well.
Fixes #251 

Note: You need to create a GitHub service connection named "vsteam" for this to work before merging/releasing the PR!!

## PR Checklist

- [ ] [Write Help](https://github.com/DarqueWarrior/vsteam/blob/master/.github/CONTRIBUTING.md#write-help)
- [ ] [Write Unit Test](https://github.com/DarqueWarrior/vsteam/blob/master/.github/CONTRIBUTING.md#write-unit-test)
- [ ] [Update CHANGELOG.md](https://github.com/DarqueWarrior/vsteam/blob/master/.github/CONTRIBUTING.md#add-a-format-file)

(not needed)

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -292,3 +292,15 @@ stages:
...
               publishFeedCredentials: 'PowerShell Gallery'
+              
+          - task: GithubRelease@0 
+            displayName: 'Create GitHub Release'      
+            inputs:
+              gitHubConnection: vsteam
+              repositoryName: DarqueWarrior/vsteam
+              tagSource: manual
+              tag: v$(PACKAGE_VERSION)      
+              addChangeLog: true
```
Comment (SebastianSchuetze) [inline azure-pipelines.yml]: @MichelZ how would that work? Which changelogs are being added here? Are those the once from git history?

Comment (MichelZ) [reply azure-pipelines.yml]: Yes.

From the docs:


> Add changelog | (Optional) If set to true,  a list of changes (commits and issues) between this and the last  published release will be generated and appended to release notes.

https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/github-release?view=azure-devops

**Comment by SebastianSchuetze** [comment]
@DarqueWarrior did you create the gitHubConnection? The last status of this PR is still "failed" due to a missing connection.

**Comment by MichelZ** [comment]
@SebastianSchuetze He adjusted the connection name in `master`, I think it's OK

**Comment by DarqueWarrior** [comment]
The GitHub release number did not expand correctly. It was v$(PACKAGE_VERSION)  and I had to manually update it.

**Comment by DarqueWarrior** [comment]
I will work through it.

**Comment by DarqueWarrior** [comment]
I see what is going on. Variables don't cross Stages.  I will work on a parsing the package version in the release.

**Comment by tomkerkhove** [comment]
> I see what is going on. Variables don't cross Stages. I will work on a parsing the package version in the release.

You can also share them cross stages, that's what I do on https://github.com/tomkerkhove/promitor.

See: 
- Version determination: https://github.com/tomkerkhove/promitor/blob/master/build/azure-devops/agents-scraper-release-official.yml#L24
- Persist version and publish as artifact: https://github.com/tomkerkhove/promitor/blob/master/build/azure-devops/agents-scraper-release-official.yml#L31

- Download and read variable from previous stage: https://github.com/tomkerkhove/promitor/blob/master/build/azure-devops/agents-scraper-release-official.yml#L44

**Comment by MichelZ** [comment]
Thanks!

**Comment by tomkerkhove** [comment]
@MichelZ & @DarqueWarrior A PR is open for that at https://github.com/DarqueWarrior/vsteam/pull/270 if you are interested.

⚠ Haven't tested the template paths since I can't run it but should be ok.

**Pull request merged**